docs/DeveloperGuide.adoc

```diff
@@ -345,6 +345,210 @@ image::UndoRedoActivityDiagram.png[width="200"]
+**Alternative 1 (current choice):** Using JDialog +
+**Pros:** JDialog does not create an application bar in the task bar +
+**Cons:** JDialog does not come with default buttons for maximizing or minimizing +
+**Alternative 2:** Using JFrame +
+**Pros:** JFrame comes with default buttons to close, maximize and minimize etc. +
+**Cons:** JFrame will create an application bar in the task bar (ugly)
+
+---
+
+**Aspect:** Implementation of `BirthdayPopup` timer +
+**Alternative 1 (current choice):** Use a timer to close the popup after 5 seconds +
+**Pros:** Closes the dialog automatically and duration of pause can be changed +
+**Cons:** The user may not get a chance to see the popup +
+**Alternative 2:** Don't use a timer +
+**Pros:** The user determines when to close it and gets a chance to see the popup +
+**Cons:** The user has to manually close it. Maybe be troublesome. +
+**Additional Information:** Planning to implement a timer reset when the mouse hovers above the popup
+
+---
+
+**Aspect:** Size of popup +
+**Alternative 1 (current choice):** Size of popup depends on the number of birthdays +
+**Pros:** The message will not be lost and it can contain any amount of names +
+**Cons:** The popup may get very large +
+**Alternative 2:** Fixed size for popup +
+**Pros:** Very neat and consistent +
+**Cons:** If the list of names gets too big, some names will not be shown +
+
+---
+// end::BirthdayNotifier[]
+
+// tag::Fuzzyfinder[]
+=== Fuzzyfinder
+
+The 'fuzzyfind' command largely operates like the 'find' command, with the caveat that it searches all substrings of contact names, not just words of contact names. This can be illustrated with the following example:
+
+    AddressBook contains: Simon, Jim, Chrissi
+    User searches for: Si
+    ---------------------
+    'find' command: returns nothing
+    'fuzzyfind' command: returns Simon, Chrissi (both contain 'si')
+
+Fuzzyfind differs from Find by the predicate being used to test each user input string with each AddressBook Name string. In this case, a new Predicate Subclass called 'NameContainsSubstringsPredicate' was created that uses 'string.contains()' instead of 'string.equals()' to check for the existence of substrings.
+
+Firstly the FuzzfindCommandParser breaks up the user input string by whitespaces and stores each smaller string in a string array. It then initializes a new FuzzyfindCommand object with the aforementioned 'NameContainsSubstringsPredicate' attribute. This results in the predicate being called during the '.execute()' method of the FuzzyCommand class, which will in turn update FilteredList of the current model object. Thus displaying all the users that match the predicate.
+
+Note that a substring match can return a lot more unexpected results than a strict word match, so it is best used if the 'find' command does not return the person you are looking for, or if you cannot remember the exact name of the person.
+
+---
+// end::Fuzzyfinder[]
```

Comment:
Would be nice to include some diagrams to explain your implementations better here (e.g. Class Diagram, Sequence Diagram, etc)